STM32F405RGTx/Core/Src/stm32f4xx_it.c

```diff
@@ -207,6 +208,20 @@ void TIM3_IRQHandler(void)
   /* USER CODE END TIM3_IRQn 1 */
 }
 
+/**
+  * @brief This function handles TIM7 global interrupt.
+  */
+void TIM7_IRQHandler(void)
+{
+  /* USER CODE BEGIN TIM7_IRQn 0 */
+
+  /* USER CODE END TIM7_IRQn 0 */
+  HAL_TIM_IRQHandler(&htim7);

```

Comment:
if we want to make the timer ISR slightly more efficient, we could call it in here, replacing the `HAL_TIM_IRQHandler`, since the the `HAL_TIM_IRQHandler` has more overhead. On the other hand, it would make the code less readable.

I honestly don't know, it's up to you.